apps/client/src/common/hooks/useSocket.ts

```diff
@@ -105,11 +105,11 @@ export const useExtraTimerControl = () => {
 };
 
 export const setExtraTimer = {
-  start: () => socketSendJson('extratimer', SimplePlayback.Start),
-  pause: () => socketSendJson('extratimer', SimplePlayback.Pause),
-  stop: () => socketSendJson('extratimer', SimplePlayback.Stop),
-  setDirection: (direction: SimpleDirection) => socketSendJson('extratimer', { direction }),
-  setTime: (time: number) => socketSendJson('extratimer', { settime: time }),
+  start: () => socketSendJson('extratimer', { 0: { playback: SimplePlayback.Start } }),
```

Comment:
I wonder if this significantly more complex, both for the implementation and consumer, without much gain.

having said that, i understand that the current api is not extensible the way we want it to be.
How about simply extratimer1 extratimer2 … ?

if we don’t find an instantiated timer at that array position we just ignore the message

Reply:
yeah that looks fine to me

how would the structure in the server be set up?
https://github.com/cpvalente/ontime/pull/854/files#diff-f75b0837f76bb68749074b654c112fa3bacadfbde32581a627983ebe0dc077e3

Reply:
I think the structure is pretty straightforward. 

The external timer service keeps an array of timers 

The adapter prepares the data object and sends it to the external timer service

The service just needs to compare the index against its array of timers and apply the call to the correct instance

Reply:
is this the form you are thinking of?
`start: () => socketSendJson('extratimer', { extratimer1: { playback: SimplePlayback.Start } }),`

Reply:
No. I meant that the address would encode the timer index.

I am not following why so we need to maintain all the messages at the same address?